platforms/ios/framework/src/TGMapView.mm

```diff
@@ -1022,15 +1050,15 @@ - (void)respondToPanGesture:(UIPanGestureRecognizer *)panRecognizer
 
     switch (panRecognizer.state) {
         case UIGestureRecognizerStateBegan:
-            [self regionWillChangeAnimated:YES];
+            [self setMapRegionChangeState:TGMapRegionAnimating];
```

Comment:
This initially seemed like a good way to categorize these events, but it turns out that MapKit actually calls `regionWillChangeAnimated: false` at the start of panning and other gestures. To match this behavior we'll need to modify our state machine a little. I think we can just add a transition from "jumping" to itself, and call `regionIsChanging` on that transition. Then for gesture events like this we'll set the state to "jumping" repeatedly.

Reply:
Yeah, works great. Good suggestion.